contracts/NodeOperatorRegistry.sol

```diff
@@ -0,0 +1,642 @@
+    /// @notice Get validator id by user address.
+    /// @param _validatorId validatorId.
+    /// @return Returns the validator total staked.
+    function validatorStake(uint256 _validatorId)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return IStakeManager(state.stakeManager).validatorStake(_validatorId);
+    }
+
+    /// @notice Get validator total stake.
+    /// @param _user user address.
+    /// @return Returns the validatorId of an address.
+    function getValidatorId(address _user)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return IStakeManager(state.stakeManager).getValidatorId(_user);
+    }
+
+    /// @notice Get validatorShare contract address.
+    /// @dev Get validatorShare contract address.
+    /// @param _validatorId Validator Id
+    /// @return Returns the address of the validatorShare contract.
+    function getValidatorContract(uint256 _validatorId)
+        external
+        view
+        override
+        returns (address)
+    {
+        return
+            IStakeManager(state.stakeManager).getValidatorContract(
+                _validatorId
+            );
+    }
+
+    /// @notice Allows to withdraw rewards from the validator.
+    /// @dev Allows to withdraw rewards from the validator using the _validatorId. Only the
+    /// owner can request withdraw in this the owner is this contract. This  functions is called
+    /// by a lido contract.
+    function withdrawRewards()
+        external
+        override
+        whenNotPaused
+        returns (uint256[] memory, address[] memory)
+    {
+        require(msg.sender == state.lido, "Caller is not the lido contract");
+        uint256[] memory shares = new uint256[](state.totalStakedNodeOpearator);
+        address[] memory recipient = new address[](
+            state.totalStakedNodeOpearator
+        );
+        uint256 index = 0;
+        uint256 totalRewards = 0;
+
+        // withdraw validator rewards
+        for (uint256 idx = 0; idx < operatorIds.length; idx++) {
+            Operator.NodeOperator memory no = operators[operatorIds[idx]];
+            if (no.status == Operator.NodeOperatorStatus.STAKED) {
+                uint256 rewards = IValidator(no.validatorContract)
+                    .withdrawRewards(no.validatorId);
+
+                recipient[index] = no.rewardAddress;
+                shares[index] = rewards;
+                totalRewards += rewards;
+                index++;
+            }
+        }
+
+        // calculate validators share
+        for (uint256 idx = 0; idx < shares.length; idx++) {
+            uint256 share = (shares[idx] * 100) / totalRewards;
+            shares[idx] = share;
+        }
+
+        emit WithdrawRewards();
+
+        return (shares, recipient);
+    }
+
+    /// @notice Allows to update signer publickey
+    /// @param _signerPubkey new signer publickey
+    function updateSigner(bytes memory _signerPubkey)
```

Comment:
Is everybody allowed to call this function?

Reply:
Yes, only the owner of each operator can update the signer publickey.